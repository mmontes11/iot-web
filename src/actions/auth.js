import {
  IS_AUTH,
  SET_USERNAME,
  SET_PASSWORD,
  LOGIN_REQUEST,
  LOGIN_REQUEST_SUCCESS,
  LOGIN_REQUEST_ERROR
} from "constants/actionTypes/auth";
import iotClient from "lib/iotClient";

export const isAuth = () => dispatch => {
  iotClient.authService
    .isAuth()
    .then(isAuthVal => dispatch({ type: IS_AUTH, isAuth: isAuthVal }))
    .catch(() => dispatch({ type: IS_AUTH, isAuth: false }));
};

export const setUsername = username => dispatch => {
  dispatch({ type: SET_USERNAME, username });
};

export const setPassword = password => dispatch => {
  dispatch({ type: SET_PASSWORD, password });
};

export const login = () => (dispatch, getState) => {
  dispatch({ type: LOGIN_REQUEST });
  const { username, password } = getState().auth;
  iotClient.authService.setCredentials(username, password);
  iotClient.authService
    .getToken()
    .then(res =>
      dispatch({ type: LOGIN_REQUEST_SUCCESS, statusCode: res.statusCode })
    )
    .catch(err =>
      dispatch({ type: LOGIN_REQUEST_ERROR, statusCode: err.statusCode, err })
    )
    .finally(() => isAuth()(dispatch));
};
