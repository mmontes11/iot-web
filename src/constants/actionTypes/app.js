export const TOGGLE_HAMBURGER_MENU = "TOGGLE_HAMBURGER_MENU";
export const TOGGLE_MAP_DIALOG = "TOGGLE_MAP_DIALOG";
