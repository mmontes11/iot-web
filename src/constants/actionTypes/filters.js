export const FILTER_TYPE_SELECT = "FILTER_TYPE_SELECT";
export const ADD_FILTER_TYPE = "ADD_FILTER_TYPE";
