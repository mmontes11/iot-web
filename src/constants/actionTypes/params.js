export const TYPE_SELECT = "TYPE_SELECT";
export const TYPE_UPDATED = "TYPE_UPDATED";
export const OBSERVATION_SELECT = "OBSERVATION_SELECT";
export const OBSERVATION_UPDATED = "OBSERVATION_UPDATED";
export const OBSERVATIONS_REQUEST = "OBSERVATIONS_REQUEST";
export const OBSERVATIONS_REQUEST_SUCCESS = "OBSERVATIONS_REQUEST_SUCCESS";
export const OBSERVATIONS_REQUEST_ERROR = "OBSERVATIONS_REQUEST_ERROR";
export const OBSERVATIONS_UPDATED = "OBSERVATIONS_UPDATED";
