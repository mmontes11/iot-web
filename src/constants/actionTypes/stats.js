export const TYPE_SELECT = "TYPE_SELECT";
export const TYPE_UPDATED = "TYPE_UPDATED";