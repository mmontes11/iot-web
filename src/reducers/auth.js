import { IS_AUTH, SET_PASSWORD, SET_USERNAME, SET_SHOW_ERROR } from "constants/actionTypes/auth";
import { RESET } from "constants/actionTypes/common";

const initialState = {
  isAuth: false,
  username: null,
  password: null,
  showError: true,
};

export default (state = initialState, { type, isAuth, username, password, showError }) => {
  switch (type) {
    case IS_AUTH:
      return { ...initialState, isAuth };
    case SET_USERNAME:
      return { ...state, username };
    case SET_PASSWORD:
      return { ...state, password };
    case SET_SHOW_ERROR:
      return { ...state, showError };
    case RESET:
      return { ...initialState, isAuth: state.isAuth };
    default:
      return state;
  }
};
