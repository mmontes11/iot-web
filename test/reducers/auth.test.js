import authReducer from "reducers/auth";
import { IS_AUTH, SET_USERNAME, SET_PASSWORD, SET_SHOW_ERROR } from "constants/actionTypes/auth";

describe("reducers/auth", () => {
  const initialState = {
    isAuth: false,
    username: null,
    password: null,
    showError: true,
  };
  it("has a default state", () => {
    expect(authReducer(undefined, { type: "@init" })).toMatchSnapshot();
  });
  it("reduces nothing", () => {
    expect(authReducer(initialState, { type: "WHATEVER" })).toMatchSnapshot();
  });
  it("reduces IS_AUTH", () => {
    expect(authReducer(initialState, { type: IS_AUTH, isAuth: true })).toMatchSnapshot();
  });
  it("reduces SET_USERNAME", () => {
    expect(authReducer(initialState, { type: SET_USERNAME, username: "USERNAME" })).toMatchSnapshot();
  });
  it("reduces SET_PASSWORD", () => {
    expect(authReducer(initialState, { type: SET_PASSWORD, password: "PASSWORD" })).toMatchSnapshot();
  });
  it("reduces SET_SHOW_ERROR", () => {
    expect(authReducer(initialState, { type: SET_SHOW_ERROR, showError: true })).toMatchSnapshot();
  });
});
