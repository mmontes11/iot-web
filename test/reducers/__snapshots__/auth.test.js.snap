// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`reducers/history has a default state 1`] = `
Object {
  "isAuth": false,
  "password": null,
  "showError": true,
  "username": null,
}
`;

exports[`reducers/history reduces IS_AUTH 1`] = `
Object {
  "isAuth": true,
  "password": null,
  "showError": true,
  "username": null,
}
`;

exports[`reducers/history reduces SET_PASSWORD 1`] = `
Object {
  "isAuth": false,
  "password": "PASSWORD",
  "showError": true,
  "username": null,
}
`;

exports[`reducers/history reduces SET_SHOW_ERROR 1`] = `
Object {
  "isAuth": false,
  "password": null,
  "showError": true,
  "username": null,
}
`;

exports[`reducers/history reduces SET_USERNAME 1`] = `
Object {
  "isAuth": false,
  "password": null,
  "showError": true,
  "username": "USERNAME",
}
`;

exports[`reducers/history reduces nothing 1`] = `
Object {
  "isAuth": false,
  "password": null,
  "showError": true,
  "username": null,
}
`;
